shared/testutil/deposits.go

```diff
@@ -57,31 +57,10 @@ func DeterministicDepositsAndKeys(numDeposits uint64) ([]*ethpb.Deposit, []bls.S
 
 		// Create the new deposits and add them to the trie.
 		for i := uint64(0); i < numRequired; i++ {
-			withdrawalCreds := hashutil.Hash(publicKeys[i+1].Marshal())
-			withdrawalCreds[0] = params.BeaconConfig().BLSWithdrawalPrefixByte
-
-			depositData := &ethpb.Deposit_Data{
-				PublicKey:             publicKeys[i].Marshal(),
-				Amount:                params.BeaconConfig().MaxEffectiveBalance,
-				WithdrawalCredentials: withdrawalCreds[:],
-			}
-
-			domain, err := helpers.ComputeDomain(params.BeaconConfig().DomainDeposit, nil, nil)
-			if err != nil {
-				return nil, nil, errors.Wrap(err, "could not compute domain")
-			}
-			root, err := ssz.SigningRoot(depositData)
-			if err != nil {
-				return nil, nil, errors.Wrap(err, "could not get signing root of deposit data")
-			}
-			sigRoot, err := (&pb.SigningData{ObjectRoot: root[:], Domain: domain}).HashTreeRoot()
+			balance := params.BeaconConfig().MaxEffectiveBalance
+			deposit, err := signedDeposit(secretKeys[i], publicKeys[i].Marshal(), publicKeys[i+1].Marshal(), balance)
```

Comment:
any check here that i+1 wont exceed the list size?

Reply:
Not required, above we generate it with a +1 as well.